webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -393,55 +405,87 @@ trait PromiseGenerationService {
+
+      _                    =  PolicyLogger.debug(s"Size of updatedNodeConfigs is ${memoryMeter.measure(updatedNodeConfigs)} ")
+      _                    =  PolicyLogger.debug(s"Size deepof updatedNodeConfigs is ${memoryMeter.measureDeep(updatedNodeConfigs)} ")
+
+      _                    =  PolicyLogger.debug(s"Size of updatedNodeInfo is ${memoryMeter.measure(updatedNodeInfo)} ")
+      _                    =  PolicyLogger.debug(s"Size deepof updatedNodeInfo is ${memoryMeter.measureDeep(updatedNodeInfo)} ")
+
+      _                    =  PolicyLogger.debug(s"Size of updatedNodesId is ${memoryMeter.measure(updatedNodesId)} ")
+      _                    =  PolicyLogger.debug(s"Size deepof updatedNodesId is ${memoryMeter.measureDeep(updatedNodesId)} ")
 
-      reportTime            =  System.currentTimeMillis
-      expectedReports       =  computeExpectedReports(ruleVals, updatedNodeConfigs.values.toSeq, updatedNodeConfigIds, generationTime, allNodeModes)
-      timeSetExpectedReport =  (System.currentTimeMillis - reportTime)
-      _                     =  PolicyLogger.debug(s"Reports updated in ${timeSetExpectedReport} ms")
 
       ///// so now we have everything for each updated nodes, we can start writing node policies and then expected reports
 
       // WHY DO WE NEED TO FORGET OTHER NODES CACHE INFO HERE ?
-      _                     <- forgetOtherNodeConfigurationState(nodeConfigs.keySet) ?~! "Cannot clean the configuration cache"
+      _                     <- forgetOtherNodeConfigurationState(allNodeConfigsId) ?~! "Cannot clean the configuration cache"
 
       writeTime             =  System.currentTimeMillis
-      writtenNodeConfigs    <- writeNodeConfigurations(rootNodeId, updatedNodeConfigIds, nodeConfigs, allLicenses, globalPolicyMode, generationTime, parallelism) ?~!"Cannot write nodes configuration"
+      _                     <- writeNodeConfigurations(rootNodeId, updatedNodeConfigIds, updatedNodeConfigs, allNodeConfigsInfos, allLicenses, globalPolicyMode, generationTime, parallelism) ?~!"Cannot write nodes configuration"
       timeWriteNodeConfig   =  (System.currentTimeMillis - writeTime)
       _                     =  PolicyLogger.debug(s"Node configuration written in ${timeWriteNodeConfig} ms, start to update expected reports.")
 
+      reportTime            =  System.currentTimeMillis
+      expectedReports       =  computeExpectedReports(updatedNodeConfigs, updatedNodeConfigIds, generationTime, allNodeModes)
+      timeSetExpectedReport =  (System.currentTimeMillis - reportTime)
+      _                     =  PolicyLogger.debug(s"Reports computed in ${timeSetExpectedReport} ms")
+
+      _                    =  PolicyLogger.debug(s"Size of expectedReports is ${memoryMeter.measure(expectedReports)} ")
+      _                    =  PolicyLogger.debug(s"Size deepof expectedReports is ${memoryMeter.measureDeep(expectedReports)} ")
```

Comment:
[2019-12-15 21:15:44] DEBUG policy.generation - Size of allNodeInfos is 24 
[2019-12-15 21:15:46] DEBUG policy.generation - Size deepof allNodeInfos is 26394576 
[2019-12-15 21:15:46] DEBUG policy.generation - Size of allNodeModes is 24 
[2019-12-15 21:15:46] DEBUG policy.generation - Size deepof allNodeModes is 1132752 
[2019-12-15 21:15:46] DEBUG policy.generation - Size of nodeConfigCaches is 16 
[2019-12-15 21:15:46] DEBUG policy.generation - Size deepof nodeConfigCaches is 16 
[2019-12-15 21:15:56] DEBUG policy.generation - Size of ruleVals is 24 
[2019-12-15 21:32:14] DEBUG policy.generation - Size deepof ruleVals is 1978959240 
[2019-12-15 21:32:33] DEBUG policy.generation - Size of nodeContexts is 24 
[2019-12-15 21:32:38] DEBUG policy.generation - Size deepof nodeContexts is 304977352 
[2019-12-15 21:35:28] DEBUG policy.generation - Size of nodeConfigs is 24 
[2019-12-15 21:56:58] DEBUG policy.generation - Size deepof nodeConfigs is 3474356896 
[2019-12-15 21:57:13] DEBUG policy.generation - Size of updatedNodeConfigIds is 24 
[2019-12-15 21:57:13] DEBUG policy.generation - Size deepof updatedNodeConfigIds is 1423528 
[2019-12-15 21:57:13] DEBUG policy.generation - Size of updatedNodeConfigs is 24 
[2019-12-15 22:18:08] DEBUG policy.generation - Size deepof updatedNodeConfigs is 3504843864 
[2019-12-15 22:18:08] DEBUG policy.generation - Size of updatedNodeInfo is 24 
[2019-12-15 22:18:09] DEBUG policy.generation - Size deepof updatedNodeInfo is 26395416 
[2019-12-15 22:18:09] DEBUG policy.generation - Size of updatedNodesId is 24 
[2019-12-15 22:18:09] DEBUG policy.generation - Size deepof updatedNodesId is 804984 
[2019-12-15 22:28:57] DEBUG policy.generation - Size of expectedReports is 24 
[2019-12-15 22:34:29] DEBUG policy.generation - Size deepof expectedReports is 241639688

Reply:
largest objects are 
updatedNodeConfigs = 3 504 843 864
nodeConfigs = 3 474 356 896
ruleVals = 1 978 959 240
nodeContexts = 304 977 352
expectedReports = 241 639 688

ping @fanf

Reply:
it's in byte or bite ?

Reply:
this is in byte

Reply:
out of the 2 668 075 048 within nodeConfigs, 2 582 780 816 are for the policies